clients/drcachesim/drpt2trace/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice,
+#   this list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of Google, Inc. nor the names of its contributors may be
+#   used to endorse or promote products derived from this software without
+#   specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL VMWARE, INC. OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+# DAMAGE.
+
+cmake_minimum_required(VERSION 3.7)
+
+include(../../../make/policies.cmake NO_POLICY_SCOPE)
+
+# Add libipt to build system. For kernel traces decoding,
+# we need to activate SIDEBAND, PEVENT and FEATURE_ELF.
+# XXX: libipt does not use the CMake export mechanism for modular separation.
+# It puts us at risk of CMake name collisions. Since libipt's build system is
+# not complicated, we can rewrite a new CMakeList.txt for it and make it modular
+# using the cmake export mechanism.
+set(SIDEBAND ON CACHE BOOL "Enable libipt-sb, a sideband correlation library." FORCE)
+set(PEVENT ON CACHE BOOL "Enable perf_event sideband support." FORCE)
+set(FEATURE_ELF ON CACHE BOOL "Support ELF files." FORCE)
+add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/libipt
+  ${PROJECT_BINARY_DIR}/third_party/libipt)
+
+# drpt2ir depends on libipt and libipt-sb. Also, the head files
+# intel-pt.h and libipt-sb.h are generated into the project binary directory.
+# So we need to add the subdirectories that contain them to the include path.
+include_directories(SYSTEM
+  ${PROJECT_BINARY_DIR}/third_party/libipt/libipt/include
+  ${PROJECT_BINARY_DIR}/third_party/libipt/sideband/include
+)
+
+# libipt.so and libipt-sb.so will be built in the project binary directory.
+# Add link directories for libipt-ext and drpt2ir to find them.
+link_directories(
+  ${PROJECT_BINARY_DIR}/lib
+)
+
+# The library libipt-ext and drpt2ir need to find headers from
+# the following directories.
+include_directories(
+  ${PROJECT_SOURCE_DIR}/third_party/libipt/ptxed/include
```

Comment:
I thought we weren't using XED?

Reply:
Yes. We don't use XED.
This path includes all header files for libipt-ext.

Reply:
The path says "ptxed": is that other things besides XED?

Reply:
This path contains one header file. The file only has one c function, which is used to splits elf sections from kcore. I plan to implement this feature using C++ in the next PR. Do you prefer to do this work in this PR?

Reply:
No, just checking that it was needed; the name makes it sound like it's for XED.